src/main/python/ttconv/imsc/style_properties.py

```diff
@@ -0,0 +1,689 @@
+      for shadow in xml_attrib.split(","):
+
+        cs = shadow.split(" ")
+
+        if len(cs) < 1 or len(cs) > 4:
+          raise ValueError("Invalid Syntax")
+        
+        x_offset = StyleProperties.ttml_length_to_model(context, cs[0])
+
+        y_offset = StyleProperties.ttml_length_to_model(context, cs[1])
+
+        blur_radius = None
+        color = None
+
+        if len(cs) == 3:
+
+          try:
+
+            blur_radius = StyleProperties.ttml_length_to_model(context, cs[2])
+
+          except ValueError:
+
+            color = utils.parse_color(cs[2])
+
+        else: # len(cs) == 4
+
+          blur_radius = StyleProperties.ttml_length_to_model(context, cs[2])
+          color = utils.parse_color(cs[3])
+
+        shadows.append(
+          styles.TextShadowType.Shadow(
+            x_offset=x_offset,
+            y_offset=y_offset,
+            blur_radius=blur_radius,
+            color=color
+          )
+        )
+
+      return styles.TextShadowType(shadows)
+
+  class UnicodeBidi(StyleProperty):
+    '''Corresponds to tts:unicodeBidi.'''
+
+    ns = "http://www.w3.org/ns/ttml#styling"
+    local_name = "unicodeBidi"
+    model_prop = styles.StyleProperties.UnicodeBidi
+
+    @classmethod
+    def extract(cls, context: StyleParsingContext, xml_attrib: str):
+      return styles.UnicodeBidiType[xml_attrib]
+
+  class Visibility(StyleProperty):
+    '''Corresponds to tts:visibility.'''
+
+    ns = "http://www.w3.org/ns/ttml#styling"
+    local_name = "visibility"
+    model_prop = styles.StyleProperties.Visibility
+    
+    @classmethod
+    def extract(cls, context: StyleParsingContext, xml_attrib: str):
+      return styles.VisibilityType[xml_attrib]
+
+  class WrapOption(StyleProperty):
+    '''Corresponds to tts:wrapOption.'''
+
+    ns = "http://www.w3.org/ns/ttml#styling"
+    local_name = "wrapOption"
+    model_prop = styles.StyleProperties.WrapOption
+
+    @classmethod
+    def extract(cls, context: StyleParsingContext, xml_attrib: str):
+      return styles.WrapOptionType[xml_attrib]
+
+  class WritingMode(StyleProperty):
+    '''Corresponds to tts:writingMode.'''
+
+    ns = "http://www.w3.org/ns/ttml#styling"
+    local_name = "writingMode"
+    model_prop = styles.StyleProperties.WritingMode
+
+    @classmethod
+    def extract(cls, context: StyleParsingContext, xml_attrib: str):
+
+      if xml_attrib == "lr":
+        return styles.WritingModeType.lrtb
+
+      if xml_attrib == "rl":
+        return styles.WritingModeType.rltb
+
+      if xml_attrib == "tb":
+        return styles.WritingModeType.tbrl
+
+      return styles.WritingModeType[xml_attrib]
+
+  BY_QNAME = {
+    f"{{{v.ns}}}{v.local_name}" : v
+    for n, v in list(locals().items()) if inspect.isclass(v)
+    }
+
+  @classmethod
+  def ttml_length_to_model(cls, _context: StyleParsingContext, xml_attrib: str):
+    (value, units) = utils.parse_length(xml_attrib)
+
+    return styles.LengthType(value, styles.LengthType.Units(units))
+    
```

Comment:
It seems Git expects a new line here

Reply:
@valnoel Git? Do you mean pylint?

Reply:
Fixed at https://github.com/sandflow/ttconv/pull/13/commits/c435c36807900c72c2042313ef7749680b71f8b2